### PR TITLE
Fix `Message` so it always adds its message

### PR DESCRIPTION
**Pull request by chengsun**
Fix to `message` issue we were discussing on gitter earlier.

Add a unit test checking for the issue.

### CODE REVIEW

File: src/combinator.rs [hunk 1]
```diff
@@ -1673,21 +1673,21 @@ impl<I, P> Parser for Message<P>
...
-                    errors.add_error(Error::Message(self.1.clone()));
-                    errors
-                })
-            })
-    }
-
     #[inline]
     fn parse_lazy(&mut self, input: I) -> ConsumedResult<Self::Output, I> {
-        self.0.parse_lazy(input.clone())
+        match self.0.parse_lazy(input.clone()) {
```
Comment (Marwes) [inline src/combinator.rs]: Unnecessary clone (it was unnecessary before as well).

**Comment by Marwes** [comment]
This was the solution I figured would work as well yesterday :) Seems like you arrived at the same solution!

**Comment by chengsun** [comment]
Great!

Removed the `clone`.

**Comment by Marwes** [comment]
Thanks. Will try and do a patch release later today.

**Comment by chengsun** [comment]
I've also spotted the same issue in `Expected`, and some other bugs, so you might want to hold off until I send you those also :)

**Comment by Marwes** [comment]
I don't think expected should add an error message when it has consumed some input though (see parsecs expected parser https://hackage.haskell.org/package/parsec-3.1.11/docs/Text-Parsec-Prim.html#v:-60--63--62-).

Consider `(expr, operator, expr).expected("binary expression").parse("1")`. Since the expression is missing an operator I'd only expected to to say `expected operator`. But if `expected` were to add an error on consumed input it will say `expected binary expression` which is wrong!

**Comment by Marwes** [comment]
That said, it would be interesting to add some way of adding contexts to errors.

`(expr, operator, expr).context("binary expression").parse("1")`

Would say something like `Expected operator in context binary expression`

**Comment by chengsun** [comment]
Let's discuss this on the PRs I'm about to open.

**Pull request merged**